envoy/tests/legacy/test_unit.py

```diff
@@ -285,25 +293,26 @@ def test_stats_prefix_optional_tags(
     fixture_file,
     metrics,
     standard_tags,
-    additional_tags,
+    optional_tags,
 ):
     instance = INSTANCES['main']
     standard_tags.append('endpoint:{}'.format(instance["stats_url"]))
-    tags_prefix = standard_tags + additional_tags
     c = check(instance)
     mock_http_response(file_path=fixture_path(fixture_file))
     dd_run_check(c)
 
-    # To ensure that this change didn't break the old behavior, both the value and the tags are asserted.
-    # The fixture is created with a specific value and the EXT_METRICS list is done in alphabetical order
-    # allowing for value to also be asserted
+    # To test the absence and presence of the optional tags, both the value and the tags are asserted.
+    # The fixtures must list all metrics, first without and then with the optional tags.
     for index, metric in enumerate(metrics):
+        # Without optional tags.
+        aggregator.assert_metric(metric, value=index, tags=standard_tags)
```

Comment:
I think the tests in CI are failing because there's only one submission for `metric`, the one with the optional tags included.

I'd suggest to run the tests locally with `ddev test envoy:py3.11-3 -- --pdb`. This will put you into a debugger session at the point of failure. Then you can inspect the [aggregator object](https://datadoghq.dev/integrations-core/base/api/#datadog_checks.base.stubs.aggregator.AggregatorStub) to find out what is being submitted.